### DISCUSSION TITLE
How to find molecular-like units from a crystal structure ?

### DISCUSSION THREAD

**Discussion by kuwabara74** [materialsproject/pymatgen#3246] [Q&A] [answered]
Do you have any idea to find molecular-like structure units from a periodic structure by pymatgen?
Here, I mean that molecular-like units are something like a phosphate ion (PO4)^3-, a formate ion (COOH)^-, a carbodiimide ion (CN2)^2-, a methylammonium ion (CH3NH3)^+ and so on.
I see the contents of func_groups.json and find this template include some functional groups but does not cover all.

If you have any idea, please let me know.

Thank you for your kind help.

**Comment by shyuep** [comment, marked as answer]
PO4 groups are not "molecular units". Pls look at the local_env package to see if there is something there that meets your needs.

**Comment by kuwabara74** [comment]
Hello @shyuep 
Thank you for your comment. My explanation was not correct. I noticed that PO4 is not a molecule. It should be called as a polyanion.
I would like to search polyanion or polycation from crystal structures.
Structure visualization software help it but it will takes long time in case of so many structures.

The local_env package seems to support what I want to do.
Thank you very much for your kind help.
